src/main/java/uk/gov/hmcts/divorce/systemupdate/schedule/conditionalorder/SystemRedoPronouncedCoverLettersTask.java

```diff
@@ -0,0 +1,123 @@
+package uk.gov.hmcts.divorce.systemupdate.schedule.conditionalorder;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.divorce.idam.IdamService;
+import uk.gov.hmcts.divorce.idam.User;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdConflictException;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdManagementException;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdSearchCaseException;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdUpdateService;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.List;
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static uk.gov.hmcts.ccd.sdk.type.YesOrNo.YES;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingFinalOrder;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.ConditionalOrderPronounced;
+import static uk.gov.hmcts.divorce.systemupdate.event.SystemResendCOPronouncedCoverLetter.SYSTEM_RESEND_CO_PRONOUNCED_COVER_LETTER;
+import static uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService.APPLICANT1_OFFLINE;
+import static uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService.APPLICANT2_OFFLINE;
+import static uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService.DATA;
+import static uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService.STATE;
+
+@Component
+@Slf4j
+/**
+ * Any cases which are in 'ConditionalOrderPronounced' state and where the applicants are offline and have contact details private
+ * 'Conditional order pronounced cover letters' should be regenerated with address on the top and sent to respective applicant.
+ */
+public class SystemRedoPronouncedCoverLettersTask implements Runnable {
+
+    @Autowired
+    private CcdUpdateService ccdUpdateService;
+
+    @Autowired
+    private CcdSearchService ccdSearchService;
+
+    @Autowired
+    private IdamService idamService;
+
+    @Autowired
+    private AuthTokenGenerator authTokenGenerator;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private CaseIdChecker caseIdChecker;
+
+    public static final String NOTIFICATION_FLAG = "coverLetterResent";
```

Comment:
Do we need to use the same flag as in [SystemResendCOPronouncedCoverLettersTask](https://github.com/hmcts/nfdiv-case-api/blob/a9e8fef88b0116e60330d6567f29920d5a81d8f9/src/main/java/uk/gov/hmcts/divorce/systemupdate/schedule/conditionalorder/SystemResendCOPronouncedCoverLettersTask.java#L60)? It gets [set after sending the regenerated cover letter here](https://github.com/hmcts/nfdiv-case-api/blob/a9e8fef88b0116e60330d6567f29920d5a81d8f9/src/main/java/uk/gov/hmcts/divorce/common/service/task/SendRegeneratedCOPronouncedCoverLetters.java#L33):
```suggestion
    public static final String NOTIFICATION_FLAG = "coPronouncedCoverLetterResent";
```

Reply:
yes, I've now checked that the flag isn't set for any of the affected cases too so it will be valid to check this for them